openeo/local/collections.py

```diff
@@ -0,0 +1,250 @@
+    data_attrs_original  = [x for x in data.attrs]
+    data_attrs = dict(zip(data_attrs_lowercase,data_attrs_original))
+    if 'title' in data_attrs_lowercase:
+        metadata['title'] = data.attrs[data_attrs['title']]
+    else:
+        metadata['title'] = file_path
+    if 'description' in data_attrs_lowercase:
+        metadata['description'] = data.attrs[data_attrs['description']]
+    else:
+        metadata['description'] = ''
+    if 'license' in data_attrs_lowercase:
+        metadata['license'] = data.attrs[data_attrs['license']]
+    else:
+        metadata['license'] = ''
+    providers = [{'name':'',
+                 'roles':['producer'],
+                 'url':''}]
+    if 'providers' in data_attrs_lowercase:
+        providers[0]['name'] = data.attrs[data_attrs['providers']]
+        metadata['providers'] = providers
+    elif 'institution' in data_attrs_lowercase:
+        providers[0]['name'] = data.attrs[data_attrs['institution']]
+        metadata['providers'] = providers
+    else:
+        metadata['providers'] = providers
+    if 'links' in data_attrs_lowercase:
+        metadata['links'] = data.attrs[data_attrs['links']]
+    else:
+        metadata['links'] = ''
+    x_min = data[x_dim].min().item(0)
+    x_max = data[x_dim].max().item(0)
+    y_min = data[y_dim].min().item(0)
+    y_max = data[y_dim].max().item(0)
+
+    crs_present = False
+    bands = list(data.data_vars)
+    if 'crs' in bands:
+        bands.remove('crs')
+        crs_present = True
+    extent = {}
+    if crs_present:
+        if 'crs_wkt' in data.crs.attrs:
+            transformer = Transformer.from_crs(data.crs.attrs['crs_wkt'], "epsg:4326")
+            lat_min,lon_min = transformer.transform(x_min,y_min)
+            lat_max,lon_max = transformer.transform(x_max,y_max)               
+            extent['spatial'] = {'bbox': [[lon_min, lat_min, lon_max, lat_max]]}
+
+    if t_dim is not None:
+        t_min = str(data[t_dim].min().values)
+        t_max = str(data[t_dim].max().values)
+        extent['temporal'] = {'interval': [[t_min,t_max]]}
+
+    metadata['extent'] = extent
+
+    t_dimension = {}
+    if t_dim is not None:
+        t_dimension = {t_dim: {'type': 'temporal', 'extent':[t_min,t_max]}}
+
+    x_dimension = {x_dim: {'type': 'spatial','axis':'x','extent':[x_min,x_max]}}
+    y_dimension = {y_dim: {'type': 'spatial','axis':'y','extent':[y_min,y_max]}}
+    if crs_present:
+        if 'crs_wkt' in data.crs.attrs:
+            x_dimension[x_dim]['reference_system'] = data.crs.attrs['crs_wkt']
+            y_dimension[y_dim]['reference_system'] = data.crs.attrs['crs_wkt']
+
+    b_dimension = {}
+    if len(bands)>0:
+        b_dimension = {'bands': {'type': 'bands', 'values':bands}}
+
+    metadata['cube:dimensions'] = {**t_dimension,**x_dimension,**y_dimension,**b_dimension}
+
+    return metadata
+
+def _get_geotiff_metadata(file_path):
+    data = rioxarray.open_rasterio(file_path.as_posix(),chunks={})
```

Comment:
it looks like there is a lot of overlap/duplication with `_get_netcdf_zarr_metadata`
Can't this be handled more generically?